core/handler/manager_server.go

```diff
@@ -367,8 +369,8 @@ func (h *handlerManager) SetApplication(ctx context.Context, in *pb.Application)
 }
 
 func (h *handlerManager) DeleteApplication(ctx context.Context, in *pb.ApplicationIdentifier) (*empty.Empty, error) {
-	if !in.Validate() {
-		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Application Identifier", "validation failed"))
+	if err := in.Validate(); err != nil {
+		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Application Identifier", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`